### PR TITLE
Feature/collect dynamic submission info

### PR DESCRIPTION
**Pull request by atuldh**

### CODE REVIEW

File: app/uk/gov/hmrc/cbcrfrontend/controllers/Submission.scala [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  }
+
+  val submitFilingType = sec.AsyncAuthenticatedAction { authContext => implicit request =>
+
+    filingTypeForm.bindFromRequest.fold(
+      formWithErrors => Future.successful(BadRequest(uk.gov.hmrc.cbcrfrontend.views.html.forms.submitInfoFilingType(
+        includes.asideBusiness(), includes.phaseBannerBeta(), formWithErrors))),
+      success => {
+        session.save(success)
+        Future.successful(Ok(uk.gov.hmrc.cbcrfrontend.views.html.forms.submitInfoUltimateParentEntity(
```
Comment (MaxWorgan) [inline app/uk/gov/hmrc/cbcrfrontend/controllers/Submission.scala]: I think we probably want to map into the session.save(success) to ensure that it successfully saves before return an OK

File: app/uk/gov/hmrc/cbcrfrontend/controllers/Submission.scala [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+
+
+  val submitUltimateParentEntity = sec.AsyncAuthenticatedAction { authContext => implicit request =>
+
+    ultimateParentEntityForm.bindFromRequest.fold(
+      formWithErrors => Future.successful(BadRequest(uk.gov.hmrc.cbcrfrontend.views.html.forms.submitInfoUltimateParentEntity(
+        includes.asideBusiness(), includes.phaseBannerBeta(), formWithErrors))),
+      success => {
+        session.save(success)
```
Comment (MaxWorgan) [inline app/uk/gov/hmrc/cbcrfrontend/controllers/Submission.scala]: see above

File: app/uk/gov/hmrc/cbcrfrontend/controllers/Submission.scala [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+
+
+  val submitFilingCapacity = sec.AsyncAuthenticatedAction { authContext => implicit request =>
+
+    filingCapacityForm.bindFromRequest.fold(
+      formWithErrors => Future.successful(BadRequest(uk.gov.hmrc.cbcrfrontend.views.html.forms.submitInfoFilingCapacity(
+        includes.asideBusiness(), includes.phaseBannerBeta(), formWithErrors))),
+      success => {
+        session.save(success)
```
Comment (MaxWorgan) [inline app/uk/gov/hmrc/cbcrfrontend/controllers/Submission.scala]: and again

File: app/uk/gov/hmrc/cbcrfrontend/controllers/Submission.scala [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+  }
+
+  val submitSubmitterInfo = sec.AsyncAuthenticatedAction { authContext => implicit request =>
+
+    submitterInfoForm.bindFromRequest.fold(
+      formWithErrors => Future.successful(BadRequest(uk.gov.hmrc.cbcrfrontend.views.html.forms.submitterInfo(
+        includes.asideBusiness(), includes.phaseBannerBeta(), formWithErrors
+      ))),
+      success => {
+        session.save(success)
```
Comment (MaxWorgan) [inline app/uk/gov/hmrc/cbcrfrontend/controllers/Submission.scala]: +1

File: app/uk/gov/hmrc/cbcrfrontend/controllers/package.scala [hunk 5]
```diff
@@ -38,4 +38,16 @@ import scala.concurrent.Future
...
+    val primary = "PRIMARY"
+    val voluntary = "VOLUNTARY"
+    val local = "LOCAL"
+  }
+
+  object FilingUserCapacity {
+    val mneUser = "MNE_USER"
+    val authorisedAgent = "AUTHORISED_AGENT"
+    val representative = "REPRESENTATIVE"
+  }
```
Comment (MaxWorgan) [inline app/uk/gov/hmrc/cbcrfrontend/controllers/package.scala]: Could I suggest using case objects and a sealed trait? It's just a little less error prone than using strings.

File: app/uk/gov/hmrc/cbcrfrontend/controllers/Submission.scala (hunk 1 above)
Comment (atuldh) [inline app/uk/gov/hmrc/cbcrfrontend/controllers/Submission.scala]: Agree

**Pull request merged**